src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java

```diff
@@ -327,4 +351,178 @@ public static OpenLoadFlowParameters get(LoadFlowParameters parameters) {
+                                                          MatrixFactory matrixFactory, Reporter reporter, boolean breakers, boolean forceA1Var) {
+        AcLoadFlowParameters acParameters = createAcParameters(parameters, parametersExt, matrixFactory, reporter, breakers, forceA1Var);
+        if (parameters.isReadSlackBus()) {
+            acParameters.getNetworkParameters().setSlackBusSelector(new NetworkSlackBusSelector(network, acParameters.getNetworkParameters().getSlackBusSelector()));
+        }
+        return acParameters;
+    }
+
+    public static AcLoadFlowParameters createAcParameters(LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt,
+                                                          MatrixFactory matrixFactory, Reporter reporter, boolean breakers, boolean forceA1Var) {
+        SlackBusSelector slackBusSelector = SlackBusSelector.fromMode(parametersExt.getSlackBusSelectionMode(), parametersExt.getSlackBusesIds());
+
+        var networkParameters = getNetworkParameters(parameters, parametersExt, slackBusSelector, breakers);
+
+        var equationSystemCreationParameters = new AcEquationSystemCreationParameters(forceA1Var);
+
+        VoltageInitializer voltageInitializer = getExtendedVoltageInitializer(parameters, parametersExt, networkParameters, matrixFactory, reporter);
+
+        var newtonRaphsonParameters = new NewtonRaphsonParameters()
+                .setVoltageInitializer(voltageInitializer)
+                .setStoppingCriteria(new DefaultNewtonRaphsonStoppingCriteria(parametersExt.getNewtonRaphsonConvEpsPerEq()))
+                .setMaxIteration(parametersExt.getMaxIteration());
+
+        OuterLoopConfig outerLoopConfig = OuterLoopConfig.findOuterLoopConfig(new DefaultOuterLoopConfig());
+        List<OuterLoop> outerLoops = outerLoopConfig.configure(parameters, parametersExt);
+
+        return new AcLoadFlowParameters(networkParameters,
+                                        equationSystemCreationParameters,
+                                        newtonRaphsonParameters,
+                                        outerLoops, matrixFactory);
+    }
+
+    public static DcLoadFlowParameters createDcParameters(Network network, LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt,
+                                                          MatrixFactory matrixFactory, boolean forcePhaseControlOffAndAddAngle1Var) {
+        var dcParameters = createDcParameters(parameters, parametersExt, matrixFactory, forcePhaseControlOffAndAddAngle1Var);
+        if (parameters.isReadSlackBus()) {
+            dcParameters.getNetworkParameters().setSlackBusSelector(new NetworkSlackBusSelector(network, dcParameters.getNetworkParameters().getSlackBusSelector()));
+        }
+        return dcParameters;
+    }
+
+    public static DcLoadFlowParameters createDcParameters(LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt,
```

Comment:
Same remark here.

Reply:
It might be used in another private project